frame/staking/src/lib.rs

```diff
@@ -1845,3 +2506,164 @@ impl<T, Reporter, Offender, R, O> ReportOffence<Reporter, Offender, O>
 		}
 	}
 }
+
+impl<T: Trait> sp_runtime::BoundToRuntimeAppPublic for Module<T> {
+	type Public = T::KeyType;
+}
+
+impl<T: Trait> pallet_session::OneSessionHandler<T::AccountId> for Module<T> {
+	type Key = T::KeyType;
+
+	fn on_genesis_session<'a, I: 'a>(validators: I)
+		where I: Iterator<Item=(&'a T::AccountId, T::KeyType)>
+	{
+		assert!(Self::keys().is_empty(), "Keys are already initialized!");
+		<Keys<T>>::put(validators.map(|x| x.1).collect::<Vec<_>>());
+	}
+
+	fn on_new_session<'a, I: 'a>(_changed: bool, validators: I, _queued_validators: I)
+		where I: Iterator<Item=(&'a T::AccountId, T::KeyType)>
+	{
+		// Update they keys
+		<Keys<T>>::put(validators.map(|x| x.1).collect::<Vec<_>>());
+	}
+
+	fn on_before_session_ending() {}
+
+	fn on_disabled(_i: usize) {}
+}
+
+
+/// Disallows any transactions that change the election result to be submitted after the election
+/// window is open.
+#[derive(Encode, Decode, Clone, Eq, PartialEq)]
+pub struct LockStakingStatus<T>(sp_std::marker::PhantomData<T>);
+
+impl<T: Trait + Send + Sync> sp_std::fmt::Debug for LockStakingStatus<T> {
+	#[cfg(feature = "std")]
+	fn fmt(&self, f: &mut sp_std::fmt::Formatter) -> sp_std::fmt::Result {
+		write!(f, "LockStakingStatus<{:?}>", self.0)
+	}
+	#[cfg(not(feature = "std"))]
+	fn fmt(&self, _: &mut sp_std::fmt::Formatter) -> sp_std::fmt::Result {
+		Ok(())
+	}
+}
+
+impl<T> LockStakingStatus<T> {
+	/// Create new `LockStakingStatus`.
+	pub fn new() -> Self {
+		Self(sp_std::marker::PhantomData)
+	}
+}
+
+impl<T> Default for LockStakingStatus<T> {
+	fn default() -> Self {
+		Self::new()
+	}
+}
+
+impl<T: Trait + Send + Sync> SignedExtension for LockStakingStatus<T> {
+	const IDENTIFIER: &'static str = "LockStakingStatus";
+	type AccountId = T::AccountId;
+	type Call = <T as Trait>::Call;
+	type AdditionalSigned = ();
+	type DispatchInfo = frame_support::weights::DispatchInfo;
+	type Pre = ();
+
+	fn additional_signed(&self) -> Result<(), TransactionValidityError> { Ok(()) }
+
+	fn validate(
+		&self,
+		_who: &Self::AccountId,
+		call: &Self::Call,
+		_info: Self::DispatchInfo,
+		_len: usize,
+	) -> TransactionValidity {
+		if let Some(inner_call) = call.is_sub_type() {
+			if let ElectionStatus::Open(_) = <Module<T>>::era_election_status() {
+				match inner_call {
```

Comment:
At the moment the number of ones that are banned are actually less so I think it is more idiomatic to have it the way it is. Also, the reverse can happen: someone adds a root function which is banned bu default. I am not sure which is better. For now, all of this is allowed:

Reply:
I am in favor of Basti suggestion, by blacklisting this can happen: we introduce a new call with a test that works, we forget to blacklist it, it opens a breach.
This situation can't happen when whitelisting, the test would fail.

Reply:
@thiolliere `payout_nominator` is also allowed, right?

Reply:
> payout_nominator is also allowed, right?

I don't think so, if its reward destination is staked then this can increase the stake of a nominator